File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> res;
        if(root == NULL) return res;
        queue<TreeNode*> q;
        q.push(root);

        while(!q.empty()){
            int len = q.size();
            TreeNode* back = q.back();
            res.push_back(back->val);
            for(int i = 0; i<len;i++){
                TreeNode* n = q.front();
                q.pop();
                if(n != NULL){
                    if(n->left) q.push(n->left);
                    if(n->right) q.push(n->right);
                }
            }
        }
        return res;


    }
};

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-1.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        dp = [-1]*(n+1)
        def dfs(i):
            if i >= n:
                return 0
            if dp[i]!=-1:
                return dp[i]
            left = dfs(i+1)+cost[i]
            right = dfs(i+2)+cost[i]
            dp[i]=min(left, right)
            return dp[i]


        return min(dfs(0),dfs(1))       